public/js/manifestPlayer.js

```diff
@@ -56,69 +56,67 @@ function addPlayer(player) {
+    container.appendChild(playerWrapper);
+
+    // Create our VideoElement using the Video Client and attach our player to it.
+    const videoEl = VideoClient.adapter.device.createVideoElement();
+    videoEl.style.objectFit = 'cover'
+    videoEl.style.width = "40%";
+    player.attachTo(videoEl);
+
+    // Finally we append our player to our playerWrapper.
+    document.getElementById("player-wrapper").appendChild(videoEl);
 
     // If we don't have a wrapper the buttons haven't been created yet, so lets create them.
     if (!wrapper) {
       // Create and append the play/pause button.
       const playerPlayButton = document.createElement("button");
-      playerPlayButton.id = "playerPlayButton";
+      playerPlayButton.id = "player-play-button";
       playerPlayButton.textContent = "Pause";
       container.appendChild(playerPlayButton);
 
       // Create and append the mute/unmute button.
       const playerMuteButton = document.createElement("button");
-      playerMuteButton.id = "playerMuteButton";
+      playerMuteButton.id = "player-mute-button";
       playerMuteButton.textContent = "Mute";
       container.appendChild(playerMuteButton);
 
       // Create and append the fullscreen button.
       const playerFullScreenButton = document.createElement("button");
-      playerFullScreenButton.id = "playerFullScreenButton";
+      playerFullScreenButton.id = "player-full-screen-button";
       playerFullScreenButton.textContent = "FullScreen";
       container.appendChild(playerFullScreenButton);
 
       // Event listeners to trigger our click handler functions when button elements are clicked.
       document
-        .getElementById("playerPlayButton")
+        .getElementById("player-play-button")
         .addEventListener("click", handlePlayerPlay);
       document
-        .getElementById("playerMuteButton")
+        .getElementById("player-mute-button")
         .addEventListener("click", handlePlayerMute);
       document
-        .getElementById("playerFullScreenButton")
+        .getElementById("player-full-screen-button")
         .addEventListener("click", handlePlayerFullScreen);
     }
-
-    // Create the Player Wrapper.
-    const playerWrapper = document.createElement("div");
-    playerWrapper.id = "playerWrapper";
-    container.appendChild(playerWrapper);
-
-    // Create our VideoElement using the Video Client and attach our player to it.
-    const videoEl = VideoClient.adapter.device.createVideoElement();
-    videoEl.style.objectFit = 'cover'
-    videoEl.style.height = "100%";
-    videoEl.style.width = "100%";
-    player.attachTo(videoEl);
-
-    // Finally we append our player to our playerWrapper.
-    document.getElementById("playerWrapper").appendChild(videoEl);
   }
 }
 
 // Function to create our VideoClient instance and call to create our player.
-function createVideoClient(manifestUrl, VideoClient, endpoint) {
+async function createVideoClient(manifestUrl, VideoClient, streamId) {
```

Comment:
We probably should make it clear that your videoclient instance can be reused for players and doesn't need to be recreated every time.

Reply:
Not sure I follow here, where/how is it getting recreated every time? I only see there being 1 player used within this demo.

Reply:
Sorry this needed more context, so usually with a manifest player implementations will have other videos etc. that you can click on to the side of the video or wherever (think of youtube and its suggested videos on the side). A user only ever needs one Video Client instance per session, so they can maintain that instance for however long the user is on the website. If they dispose of the Video Client instance and re create it there usually is issues with maintaining state. In the case of the player it is common to dispose it and recreate it many times when jumping between streams. Not sure if we should just have this in our docs instead of here though, but figured it might not be a bad note to add here since it isnt clear. TLDR: Video Client instance shouldnt be disposed, player can be disposed and recreated many times.